src/inference/hmc.jl

```diff
@@ -569,20 +644,22 @@ function HMCState(
     ∂logπ∂θ = gen_∂logπ∂θ(vi, spl, model)
     logπ = gen_logπ(vi, spl, model)
 
-    # Get the metric type.
-    metricT = getmetricT(spl.alg)
-
     # Create a Hamiltonian.
     θ_init = Vector{Float64}(spl.state.vi[spl])
-    metric = metricT(length(θ_init))
+    # Get the metric.
+    metric = as_concrete(spl.alg.metric, length(θ_init))
     h = AHMC.Hamiltonian(metric, logπ, ∂logπ∂θ)
 
     # Find good eps if not provided one
-    if iszero(spl.alg.ϵ)
-        ϵ = AHMC.find_good_stepsize(h, θ_init)
-        @info "Found initial step size" ϵ
+    if spl.alg.integrator isa DefaultIntegrator
+        if iszero(spl.alg.ϵ)
+            ϵ = AHMC.find_good_stepsize(h, θ_init)
+            @info "Found initial step size" ϵ
+        else
+            ϵ = spl.alg.ϵ
+        end
     else
-        ϵ = spl.alg.ϵ
+        ϵ = AHMC.step_size(spl.alg.integrator)
     end
```

Comment:
It seems this could be refactored into a separate function since it's the same as above?